### PR TITLE
Extclient NET-63x

### PR DESCRIPTION
**Pull request by mattkasun**
## Describe your changes
update Extclient models with additional editable fields
add length check to extclient names
updates to create/update extclient for additional fields
updates to nmctl 

## Provide testing steps
see NET-63x
## Checklist before requesting a review
- [ ] My changes affect only 10 files or less.
- [ ] I have performed a self-review of my code and tested it.
- [ ] If it is a new feature, I have added thorough tests, my code is <= 1450 lines.
- [ ] If it is a bugfix, my code is <= 200 lines.
- [ ] My functions are <= 80 lines.
- [ ] I have had my code reviewed by a peer.
- [ ] My unit tests pass locally.
- [ ] Netmaker is awesome.

### CODE REVIEW

**Review by theguy951357 — APPROVED** [review]
pr steps pass
critical list passes

File: cli/cmd/ext_client/create.go [hunk 1]
```diff
@@ -4,23 +4,39 @@ import (
...
+	"github.com/gravitl/netmaker/models"
 	"github.com/spf13/cobra"
 )
 
-var extClientID string
+var (
+	extClientID string
+	publicKey   string
+	dns         string
+	allowedips  []string
```
Comment (TobiaszCudnik) [inline cli/cmd/ext_client/create.go:15]: `allowedIPs`?

**Review by TobiaszCudnik — APPROVED** [review]

File: controllers/regex_test.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
+package controller
+
+import "testing"
+
+func TestValidName(t *testing.T) {
```
Comment (abhishek9686) [inline controllers/regex_test.go:5]: missing comment

File: models/node.go [hunk 3]
```diff
@@ -69,6 +69,7 @@ type CommonNode struct {
 	IsEgressGateway     bool          `json:"isegressgateway" yaml:"isegressgateway"`
 	EgressGatewayRanges []string      `json:"egressgatewayranges" bson:"egressgatewayranges" yaml:"egressgatewayranges"`
 	IsIngressGateway    bool          `json:"isingressgateway" yaml:"isingressgateway"`
+	IngressDNS          string        `json:"extclientdns" yaml:"extclientdns"`
```
Comment (abhishek9686) [inline models/node.go:72]: json tag supposed to be `ingress_dns`?

**Review by abhishek9686 — CHANGES_REQUESTED** [review]

File: controllers/ext_client.go [hunk 4]
```diff
@@ -479,25 +475,24 @@ func updateExtClient(w http.ResponseWriter, r *http.Request) {
...
 	}
 	// == END PRO ==
 
-	var changedEnabled = (newExtClient.Enabled != oldExtClient.Enabled) || // indicates there was a change in enablement
-		len(newExtClient.ACLs) != len(oldExtClient.ACLs)
+	var changedEnabled = (update.Enabled != oldExtClient.Enabled) // indicates there was a change in enablement
 	// extra var need as logic.Update changes oldExtClient
 	currentClient := oldExtClient
-	newclient, err := logic.UpdateExtClient(newExtClient.ClientID, params["network"], newExtClient.Enabled, &oldExtClient, newExtClient.ACLs)
+	newclient, err := logic.UpdateExtClient(&oldExtClient, &update)
```
[the remark this answers is not in the document]
Comment (mattkasun) [reply controllers/ext_client.go:487]: the same way there are now ee.logic.ext_acls.go

**Review by Aceix — CHANGES_REQUESTED** [review]
can you add `extclientdns` to `/${networkId}/${nodeId}/createingress`

**Comment by mattkasun** [comment]
> can you add `extclientdns` to `/${networkId}/${nodeId}/createingress`
added

**Review by Aceix — APPROVED** [review]

File: cli/cmd/ext_client/create.go [hunk 5]
```diff
@@ -4,23 +4,39 @@ import (
...
 
 var extClientCreateCmd = &cobra.Command{
 	Use:   "create [NETWORK NAME] [NODE ID]",
 	Args:  cobra.ExactArgs(2),
 	Short: "Create an External Client",
 	Long:  `Create an External Client`,
 	Run: func(cmd *cobra.Command, args []string) {
-		functions.CreateExtClient(args[0], args[1], extClientID)
+		extClient := models.CustomExtClient{
+			ClientID:        extClientID,
```
Comment (TobiaszCudnik) [inline cli/cmd/ext_client/create.go:25]: Are these simply empty string assignments?

**Review by TobiaszCudnik — APPROVED** [review]

**Comment by afeiszli** [comment]
Is this ready to merge?

**Pull request merged**